src/fragments/start/getting-started/react/auth.mdx

```diff
@@ -39,39 +56,133 @@ import { withAuthenticator, Button, Heading } from '@aws-amplify/ui-react';
 import '@aws-amplify/ui-react/styles.css';
 ```
 
-2. Wrap your application with the `withAuthenticator` component.
+2. Pass `{signOut, user}` to the `App` component:
 
 ```javascript
 /* src/App.js */
-function App({ signOut, user }) {
-  // Todo logic here
-
-  return (
-    <>
-      {/* Add Todo JSX here  */}
-      <Heading level={1}>Hello {user.username}</Heading>
-      <Button onClick={signOut}>Sign out</Button>
-    </>
-  );
+function App({ signOut, user }) { 
+  // ... 
 }
+```
+
+3. Add this heading and button block to the top of your `App` component:
+
+```javascript
+// ...
+return (
+  <div style={styles.container}>
+    <Heading level={1}>Hello {user.username}</Heading>
+    <Button onClick={signOut}>Sign out</Button>
+    <h2>Amplify Todos</h2>
+//...
+```
+
+4. Lastly, wrap your `App` export with the `withAuthenticator` Amplify UI component:
 
+```javascript
 export default withAuthenticator(App);
 ```
 
-Run the app to see the new Authentication flow protecting the app:
+Run the app to see the new authentication flow protecting the app:
 
 ```bash
 npm start
 ```
 
-Now you should see the app load with an authentication flow allowing users to sign up and sign in.
+Now you should see the app load with an authentication flow allowing users to sign up and sign in. Here's all the code below:
+
+```javascript
+/* src/App.js */
+import React, { useEffect, useState } from 'react'
+import { Amplify, API, graphqlOperation } from 'aws-amplify'
+import { createTodo } from './graphql/mutations'
+import { listTodos } from './graphql/queries'
+import { withAuthenticator, Button, Heading } from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+
+import awsExports from "./aws-exports";
+Amplify.configure(awsExports);
+
+const initialState = { name: '', description: '' }
+
+const App = ({ signOut, user }) => {
+  const [formState, setFormState] = useState(initialState)
+  const [todos, setTodos] = useState([])
+
+  useEffect(() => {
+    fetchTodos()
+  }, [])
+
+  function setInput(key, value) {
+    setFormState({ ...formState, [key]: value })
+  }
+
+  async function fetchTodos() {
+    try {
+      const todoData = await API.graphql(graphqlOperation(listTodos))
+      const todos = todoData.data.listTodos.items
+      setTodos(todos)
+    } catch (err) { console.log('error fetching todos') }
+  }
+
+  async function addTodo() {
+    try {
+      if (!formState.name || !formState.description) return
+      const todo = { ...formState }
+      setTodos([...todos, todo])
+      setFormState(initialState)
+      await API.graphql(graphqlOperation(createTodo, {input: todo}))
+    } catch (err) {
+      console.log('error creating todo:', err)
+    }
+  }
+
+  return (
+    <div style={styles.container}>
+      <Heading level={1}>Hello {user.username}</Heading>
+      <Button onClick={signOut}>Sign out</Button>
+      <h2>Amplify Todos</h2>
+      <input
```

Comment:
Thoughts on using the other Amplify primitives (`Heading` , `TextField`, etc) here for the `h2`, `input`, and `button`?

Reply:
+1

Reply:
Not a bad idea, but that would mean explaining that's what's going on. Could overload the lesson with new concepts

Reply:
The components are already imported and used: 

```html
<Heading level={1}>Hello {user.username}</Heading>
<Button onClick={signOut}>Sign out</Button>
```

I think it adds a bit more confusion to have them mixed since the names become overloaded.

Reply:
Sure, I added a bonus section